provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,455 @@
+ */
+@SuppressWarnings("WeakerAccess")
+@RestController
+@RequestMapping(
+  value = {"/api/Location"},
+  produces = {"application/json", "application/json+fhir", "application/fhir+json"}
+)
+public class LocationController {
+
+  private Transformer transformer;
+
+  private PpmsClient ppmsClient;
+
+  private Bundler bundler;
+
+  private int totalRecords;
+
+  /** Controller setup. */
+  public LocationController(
+      @Autowired LocationController.Transformer transformer,
+      @Autowired Bundler bundler,
+      @Autowired PpmsClient ppmsClient) {
+    this.transformer = transformer;
+    this.bundler = bundler;
+    this.ppmsClient = ppmsClient;
+  }
+
+  private Location.Bundle bundle(MultiValueMap<String, String> parameters, int page, int count) {
+    List<LocationWrapper> root = search(parameters, page, count);
+    LinkConfig linkConfig =
+        LinkConfig.builder()
+            .path("Location")
+            .queryParams(parameters)
+            .page(page)
+            .recordsPerPage(count)
+            .totalRecords(totalRecords)
+            .build();
+    return bundler.bundle(
+        BundleContext.of(linkConfig, root, transformer, Location.Entry::new, Location.Bundle::new));
+  }
+
+  /** Read by identifier. */
+  @GetMapping(value = {"/{publicId}"})
+  public Location readByIdentifier(@PathVariable("publicId") String publicId) {
+    return transformer.apply(search(Parameters.forIdentity((publicId)), 1, 1).get(0));
+  }
+
+  private List<LocationWrapper> search(
+      MultiValueMap<String, String> parameters, int page, int count) {
+    LocationWrapper.LocationWrapperBuilder locationWrapper =
+        LocationWrapper.builder()
+            .careSitesResponse(CareSitesResponse.builder().build())
+            .providerResponse(ProviderResponse.builder().build())
+            .providerServicesResponse(ProviderServicesResponse.builder().build());
+    List<LocationWrapper> filteredResults = new ArrayList<>();
+    List<ProviderServicesResponse> providerServicesResponsePages;
+    List<ProviderResponse.Value> providerResponsePages;
+    List<CareSitesResponse.Value> careSiteResponsePages;
+    LocationWrapper currentPage;
+    totalRecords = 1;
+    int fromIndex;
+    int toIndex;
+    if (parameters.get("identifier") != null) {
+      String identifier = parameters.get("identifier").toArray()[0].toString();
+      locationWrapper.providerServicesResponse(ppmsClient.providerServicesById(identifier));
+      locationWrapper.providerResponse(
+          ProviderResponse.builder()
+              .value(
+                  Collections.singletonList(
+                      ProviderResponse.Value.builder()
+                          .providerIdentifier(
+                              Integer.parseInt(
+                                  parameters.get("identifier").toArray()[0].toString()))
+                          .build()))
+              .build());
+      if (locationWrapper.build().providerServicesResponse().value() == null
```

Comment:
All of the PPMS response objects have lazy-getters for their collection methods, so it's not necessary to check them for `null`.